{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "66fc790a-4f68-478f-8ae8-aff536552c16",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: biopython in c:\\users\\admin\\anaconda3\\lib\\site-packages (1.84)\n",
      "Requirement already satisfied: numpy in c:\\users\\admin\\anaconda3\\lib\\site-packages (from biopython) (1.26.4)\n"
     ]
    }
   ],
   "source": [
    "!pip install biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "35e66297-6a91-4b7b-a531-a11c77ebe16c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_002755.2 Mycobacterium tuberculosis CDC1551, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_008769.1 Mycobacterium tuberculosis variant bovis BCG str. Pasteur 1173P2, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_012207.1 Mycobacterium tuberculosis variant bovis BCG str. Tokyo 172, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_009525.1 Mycobacterium tuberculosis H37Ra, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_009565.1 Mycobacterium tuberculosis F11, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_012943.1 Mycobacterium tuberculosis KZN 1435, complete sequence [96, 110, 198, 211, 1139, 1248, 1769]\n",
      "NC_022350.1 Mycobacterium tuberculosis str. Haarlem, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_016768.1 Mycobacterium tuberculosis KZN 4207, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_018078.1 Mycobacterium tuberculosis KZN 605, complete sequence [96, 110, 198, 211, 1249, 2907]\n",
      "NZ_CP012090.1 Mycobacterium tuberculosis W-148 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP003494.1 Mycobacterium tuberculosis variant bovis BCG str. ATCC 35743 strain Tice chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_002945.4 Mycobacterium tuberculosis variant bovis AF2122/97 chromosome Mycobacterium_bovis_AF212297, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_000962.3 Mycobacterium tuberculosis H37Rv, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_017524.1 Mycobacterium tuberculosis CTRI-2, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_016804.1 Mycobacterium tuberculosis variant bovis BCG str. Mexico, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_015758.1 Mycobacterium tuberculosis variant africanum GM041182, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_017523.1 Mycobacterium tuberculosis CCDC5079, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_017522.1 Mycobacterium tuberculosis CCDC5180, complete sequence [96, 110, 198, 211, 1139, 1249, 1768]\n",
      "NC_018143.2 Mycobacterium tuberculosis H37Rv, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_020089.1 Mycobacterium tuberculosis 7199-99, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_020245.2 Mycobacterium tuberculosis variant bovis BCG str. Korea 1168P, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_020559.1 Mycobacterium tuberculosis str. Erdman = ATCC 35801 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_021054.1 Mycobacterium tuberculosis str. Beijing/NITR203, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_021192.1 Mycobacterium tuberculosis str. Haarlem/NITR202, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_021193.1 Mycobacterium tuberculosis CAS/NITR204, complete genome [96, 110, 198, 211, 241, 896, 1248, 1768]\n",
      "NC_021194.1 Mycobacterium tuberculosis EAI5/NITR206, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_021251.1 Mycobacterium tuberculosis CCDC5079, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NC_021740.1 Mycobacterium tuberculosis EAI5, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP002871.1 Mycobacterium tuberculosis HKBS1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP002882.1 Mycobacterium tuberculosis BT2 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP002883.1 Mycobacterium tuberculosis BT1 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP002885.1 Mycobacterium tuberculosis CCDC5180 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP007803.1 Mycobacterium tuberculosis K chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP007809.1 Mycobacterium tuberculosis strain KIT87190 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP009100.1 Mycobacterium tuberculosis strain ZMC13-264 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP009101.1 Mycobacterium tuberculosis strain ZMC13-88 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP009426.1 Mycobacterium tuberculosis strain 96075 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP009427.1 Mycobacterium tuberculosis strain 96121 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_HG813240.1 Mycobacterium tuberculosis 49-02 chromosome I, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP007027.1 Mycobacterium tuberculosis H37RvSiena chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP014573.1 Mycobacterium tuberculosis str. Kurono chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP015773.2 Mycobacterium tuberculosis variant bovis strain SP38 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010873.1 Mycobacterium tuberculosis strain Beijing-like, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_AM412059.1 Mycobacterium tuberculosis variant bovis BCG str. Moreau RDJ isolate SL21 FAP RJ Passage B8S2 vaccine culture chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP008744.1 Mycobacterium tuberculosis variant bovis BCG strain 3281 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP012095.1 Mycobacterium tuberculosis variant bovis strain 1595 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP009243.1 Mycobacterium tuberculosis variant bovis BCG strain Russia 368 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP012506.2 Mycobacterium tuberculosis strain SCAID 187.0 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP013741.1 Mycobacterium tuberculosis variant bovis strain BCG-1 (Russia) chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010329.1 Mycobacterium tuberculosis strain F1, complete genome [96, 110, 198, 211, 1139, 1248, 1767]\n",
      "NZ_CP010330.1 Mycobacterium tuberculosis strain F28 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010335.1 Mycobacterium tuberculosis strain 2242, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010336.1 Mycobacterium tuberculosis strain 2279, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010337.1 Mycobacterium tuberculosis strain 22115 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010338.1 Mycobacterium tuberculosis strain 37004, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010339.1 Mycobacterium tuberculosis strain 22103 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP010340.1 Mycobacterium tuberculosis strain 26105 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP014566.1 Mycobacterium tuberculosis variant bovis BCG str. Tokyo 172 chromosome, complete genome [96, 110, 198, 211, 1138, 1248, 1768]\n",
      "NZ_CP046529.1 Mycobacterium tuberculosis strain SIT745/EAI1-MYS chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP016794.1 Mycobacterium tuberculosis strain SCAID 320.0 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP016888.1 Mycobacterium tuberculosis strain SCAID 252.0 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP011510.1 Mycobacterium tuberculosis strain Beijing chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP013475.1 Mycobacterium tuberculosis strain 1458 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP017920.1 Mycobacterium tuberculosis strain TB282 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018303.1 Mycobacterium tuberculosis strain I0004241-1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018305.1 Mycobacterium tuberculosis strain M0018684-2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018302.1 Mycobacterium tuberculosis strain I0004000-1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018301.1 Mycobacterium tuberculosis strain I0002801-4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018300.1 Mycobacterium tuberculosis strain I0002353-6 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018304.1 Mycobacterium tuberculosis strain M0002959-6 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP018778.1 Mycobacterium tuberculosis strain DK9897 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP016972.1 Mycobacterium tuberculosis H37Ra chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP020381.2 Mycobacterium tuberculosis strain MTB1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP017593.1 Mycobacterium tuberculosis strain Beijing-like/35049 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP017594.1 Mycobacterium tuberculosis strain Beijing-like/36918 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP017595.1 Mycobacterium tuberculosis strain Beijing-like/38774 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP017596.1 Mycobacterium tuberculosis strain Beijing/391 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP017597.1 Mycobacterium tuberculosis strain Beijing-like/50148 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP017598.1 Mycobacterium tuberculosis strain Beijing-like/1104 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP022014.1 Mycobacterium tuberculosis strain MTB2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP017901.1 Mycobacterium tuberculosis strain NCGM946K2 chromosome, complete genome [96, 111, 198, 211, 1248, 2907]\n",
      "NZ_AP018033.1 Mycobacterium tuberculosis strain HN-024 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP018034.1 Mycobacterium tuberculosis strain HN-205 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP018035.1 Mycobacterium tuberculosis strain HN-321 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP018036.1 Mycobacterium tuberculosis strain HN-506 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023573.1 Mycobacterium tuberculosis strain CSV4519 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023574.1 Mycobacterium tuberculosis strain CSV4644 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023575.1 Mycobacterium tuberculosis strain CSV5769 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP023576.1 Mycobacterium tuberculosis strain CSV10399 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023577.1 Mycobacterium tuberculosis strain CSV11678 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023578.1 Mycobacterium tuberculosis strain LE486 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023579.1 Mycobacterium tuberculosis strain LE492 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023580.1 Mycobacterium tuberculosis strain LN180 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023581.1 Mycobacterium tuberculosis strain LN2358 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023582.1 Mycobacterium tuberculosis strain LN3756 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023583.1 Mycobacterium tuberculosis strain MDRDM260 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023584.1 Mycobacterium tuberculosis strain MDRDM627 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023585.1 Mycobacterium tuberculosis strain MDRDM1098 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023586.1 Mycobacterium tuberculosis strain MDRMA2491 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023587.1 Mycobacterium tuberculosis strain ME1473 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023588.1 Mycobacterium tuberculosis strain TBDM425 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023589.1 Mycobacterium tuberculosis strain TBV5000 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023590.1 Mycobacterium tuberculosis strain TBV5362 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023591.1 Mycobacterium tuberculosis strain TBV5365 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023592.1 Mycobacterium tuberculosis strain SLM036 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP023593.1 Mycobacterium tuberculosis strain SLM040 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP023594.1 Mycobacterium tuberculosis strain SLM056 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023595.1 Mycobacterium tuberculosis strain SLM060 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP023596.1 Mycobacterium tuberculosis strain SLM063 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023597.1 Mycobacterium tuberculosis strain SLM088 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP023598.1 Mycobacterium tuberculosis strain SLM100 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023599.1 Mycobacterium tuberculosis strain CSV383 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023600.1 Mycobacterium tuberculosis strain CSV3611 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023601.1 Mycobacterium tuberculosis strain CSV9577 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023602.1 Mycobacterium tuberculosis strain LE13 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023603.1 Mycobacterium tuberculosis strain LE63 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023604.1 Mycobacterium tuberculosis strain LE76 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023605.1 Mycobacterium tuberculosis strain LE79 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023606.1 Mycobacterium tuberculosis strain LE103 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023607.1 Mycobacterium tuberculosis strain LE371 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023608.1 Mycobacterium tuberculosis strain LE410 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023609.1 Mycobacterium tuberculosis strain LN55 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023610.1 Mycobacterium tuberculosis strain LN317 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023611.1 Mycobacterium tuberculosis strain LN763 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023612.1 Mycobacterium tuberculosis strain LN2978 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023613.1 Mycobacterium tuberculosis strain LN3584 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023614.1 Mycobacterium tuberculosis strain LN3588 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023615.1 Mycobacterium tuberculosis strain LN3589 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023616.1 Mycobacterium tuberculosis strain LN3668 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023617.1 Mycobacterium tuberculosis strain LN3672 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023618.1 Mycobacterium tuberculosis strain LN3695 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023619.1 Mycobacterium tuberculosis strain LN1100 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023620.1 Mycobacterium tuberculosis strain LN1856 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP023621.1 Mycobacterium tuberculosis strain LN2900 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023622.1 Mycobacterium tuberculosis strain MDRDM827 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023623.1 Mycobacterium tuberculosis strain MDRMA203 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023624.1 Mycobacterium tuberculosis strain MDRMA701 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023625.1 Mycobacterium tuberculosis strain MDRMA863 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023626.1 Mycobacterium tuberculosis strain MDRMA1565 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023627.1 Mycobacterium tuberculosis strain MDRMA2019 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023628.1 Mycobacterium tuberculosis strain MDRMA2082 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023629.1 Mycobacterium tuberculosis strain MDRMA2260 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023630.1 Mycobacterium tuberculosis strain MDRMA2441 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023631.1 Mycobacterium tuberculosis strain TBDM1506 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023632.1 Mycobacterium tuberculosis strain TBDM2189 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023633.1 Mycobacterium tuberculosis strain TBDM2444 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023634.1 Mycobacterium tuberculosis strain TBDM2487 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023635.1 Mycobacterium tuberculosis strain TBDM2489 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023636.1 Mycobacterium tuberculosis strain TBDM2699 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023637.1 Mycobacterium tuberculosis strain TBDM2717 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023638.1 Mycobacterium tuberculosis strain TBV4766 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023639.1 Mycobacterium tuberculosis strain TBV4768 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP023640.1 Mycobacterium tuberculosis strain TBV4952 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025593.1 Mycobacterium tuberculosis strain GG-111-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025594.1 Mycobacterium tuberculosis strain GG-5-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025595.1 Mycobacterium tuberculosis strain GG-20-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025600.1 Mycobacterium tuberculosis strain GG-77-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025601.1 Mycobacterium tuberculosis strain GG-90-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025604.1 Mycobacterium tuberculosis strain GG-129-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025605.1 Mycobacterium tuberculosis strain GG-134-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025608.1 Mycobacterium tuberculosis strain GG-229-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025598.1 Mycobacterium tuberculosis strain GG-37-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025603.1 Mycobacterium tuberculosis strain GG-121-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025607.1 Mycobacterium tuberculosis strain GG-186-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025596.1 Mycobacterium tuberculosis strain GG-27-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025597.1 Mycobacterium tuberculosis strain GG-36-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025599.1 Mycobacterium tuberculosis strain GG-45-11 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025602.1 Mycobacterium tuberculosis strain GG-109-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP025606.1 Mycobacterium tuberculosis strain GG-137-10 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP022704.1 Mycobacterium tuberculosis strain Beijing2014PNGD chromosome, complete genome [96, 110, 198, 211, 1140, 1248, 1769]\n",
      "NZ_CP027035.1 Mycobacterium tuberculosis variant bovis strain 2002/0476 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP022578.1 Mycobacterium tuberculosis strain WC059 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP022577.1 Mycobacterium tuberculosis strain WC078 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP030093.1 Mycobacterium tuberculosis strain RUS_B0 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP019610.1 Mycobacterium tuberculosis strain H54 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP019611.1 Mycobacterium tuberculosis strain H83 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP019612.1 Mycobacterium tuberculosis strain H107 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP019613.1 Mycobacterium tuberculosis strain H112 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP029065.1 Mycobacterium tuberculosis strain TBMENG-03 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP039851.1 Mycobacterium tuberculosis variant bovis strain Danish delta-sapM chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP039850.1 Mycobacterium tuberculosis variant bovis strain Danish 1331 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041207.1 Mycobacterium tuberculosis strain MT-0080 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP044345.1 Mycobacterium tuberculosis strain L chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP045962.1 Mycobacterium tuberculosis strain AUSMDU00018547 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP046308.1 Mycobacterium tuberculosis strain FDAARGOS_751 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP046309.1 Mycobacterium tuberculosis strain FDAARGOS_750 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP046728.2 Mycobacterium tuberculosis strain TCDC11 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP048071.1 Mycobacterium tuberculosis strain RW-TB008 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP053092.1 Mycobacterium tuberculosis strain 4860 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP054013.1 Mycobacterium tuberculosis strain FDAARGOS_757 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP054014.1 Mycobacterium tuberculosis strain FDAARGOS_756 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP043995.1 Mycobacterium tuberculosis strain 4-0041P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP043996.1 Mycobacterium tuberculosis strain 1-0072P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP043997.1 Mycobacterium tuberculosis strain 1-0066P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041788.1 Mycobacterium tuberculosis strain SEA17020030P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041789.1 Mycobacterium tuberculosis strain SEA17020028P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041790.1 Mycobacterium tuberculosis strain SEA17020024P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041792.1 Mycobacterium tuberculosis strain SEA15230P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041791.1 Mycobacterium tuberculosis strain SEA17020023P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041793.1 Mycobacterium tuberculosis strain SEA15229P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041794.1 Mycobacterium tuberculosis strain SEA15228P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041795.1 Mycobacterium tuberculosis strain SEA14333P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041796.1 Mycobacterium tuberculosis strain SEA14318P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041797.1 Mycobacterium tuberculosis strain SEA14117P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041798.1 Mycobacterium tuberculosis strain SEA13111P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041799.1 Mycobacterium tuberculosis strain SEA13020298P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041800.1 Mycobacterium tuberculosis strain SEA12334P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041801.1 Mycobacterium tuberculosis strain SEA12202P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041802.1 Mycobacterium tuberculosis strain SEA12126P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041803.1 Mycobacterium tuberculosis strain SEA11278P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1770]\n",
      "NZ_CP041804.1 Mycobacterium tuberculosis strain SEA11020092P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041805.1 Mycobacterium tuberculosis strain SEA11020068P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041806.1 Mycobacterium tuberculosis strain SEA11020057P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041807.1 Mycobacterium tuberculosis strain SEA11020038P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041808.1 Mycobacterium tuberculosis strain SEA09167P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041809.1 Mycobacterium tuberculosis strain SEA09020048P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041810.1 Mycobacterium tuberculosis strain SEA08162P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041811.1 Mycobacterium tuberculosis strain SEA08151P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041812.1 Mycobacterium tuberculosis strain SEA07020250P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041813.1 Mycobacterium tuberculosis strain SEA07010354P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041814.1 Mycobacterium tuberculosis strain SEA06535P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041815.1 Mycobacterium tuberculosis strain SEA02010036P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041816.1 Mycobacterium tuberculosis strain SEA00042P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041817.1 Mycobacterium tuberculosis strain 4-0096P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041818.1 Mycobacterium tuberculosis strain 4-0087P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041819.1 Mycobacterium tuberculosis strain 4-0077P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041820.1 Mycobacterium tuberculosis strain 4-0073P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041821.1 Mycobacterium tuberculosis strain 4-0062P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041822.1 Mycobacterium tuberculosis strain 4-0024P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041823.1 Mycobacterium tuberculosis strain 4-0019P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041824.1 Mycobacterium tuberculosis strain 4-0012P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041825.1 Mycobacterium tuberculosis strain 4-0010P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041826.1 Mycobacterium tuberculosis strain 3-0124P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1247, 1768]\n",
      "NZ_CP041827.1 Mycobacterium tuberculosis strain 3-0096P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041828.1 Mycobacterium tuberculosis strain 3-0090P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041829.1 Mycobacterium tuberculosis strain 3-0059P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041830.1 Mycobacterium tuberculosis strain 2-0068P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041831.1 Mycobacterium tuberculosis strain 2-0059P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041832.1 Mycobacterium tuberculosis strain 2-0052P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041833.1 Mycobacterium tuberculosis strain 2-0046P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041834.1 Mycobacterium tuberculosis strain 2-0043-unknownP6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041835.1 Mycobacterium tuberculosis strain 2-0034P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041836.1 Mycobacterium tuberculosis strain 2-0031P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041837.1 Mycobacterium tuberculosis strain 2-0029P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041838.1 Mycobacterium tuberculosis strain 2-0028P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041839.1 Mycobacterium tuberculosis strain 2-0023P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041840.1 Mycobacterium tuberculosis strain 2-0022P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041841.1 Mycobacterium tuberculosis strain 2-0021P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041842.1 Mycobacterium tuberculosis strain 2-0013P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041843.1 Mycobacterium tuberculosis strain 1-0168P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041844.1 Mycobacterium tuberculosis strain 1-0160P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041845.1 Mycobacterium tuberculosis strain 1-0156P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041846.1 Mycobacterium tuberculosis strain 1-0153P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041847.1 Mycobacterium tuberculosis strain 1-0149P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041848.1 Mycobacterium tuberculosis strain 1-0137P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041849.1 Mycobacterium tuberculosis strain 1-0123P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041850.1 Mycobacterium tuberculosis strain 1-0116P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041851.1 Mycobacterium tuberculosis strain 1-0112P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041852.1 Mycobacterium tuberculosis strain 1-0110P6C4 chromosome, complete genome [96, 109, 198, 211, 1248, 2907]\n",
      "NZ_CP041853.1 Mycobacterium tuberculosis strain 1-0107P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041854.1 Mycobacterium tuberculosis strain 1-0102P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041855.1 Mycobacterium tuberculosis strain 1-0084P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041856.1 Mycobacterium tuberculosis strain 1-0071P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041857.1 Mycobacterium tuberculosis strain 1-0069P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041858.1 Mycobacterium tuberculosis strain 1-0064P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041859.1 Mycobacterium tuberculosis strain 1-0061P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041860.1 Mycobacterium tuberculosis strain 1-0056P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041861.1 Mycobacterium tuberculosis strain 1-0054P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041862.1 Mycobacterium tuberculosis strain 1-0047P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041863.1 Mycobacterium tuberculosis strain 1-0045P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041864.1 Mycobacterium tuberculosis strain 1-0044P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041865.1 Mycobacterium tuberculosis strain 1-0039P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041866.1 Mycobacterium tuberculosis strain 1-0036P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041867.1 Mycobacterium tuberculosis strain 1-0031P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041868.1 Mycobacterium tuberculosis strain 1-0030P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041869.1 Mycobacterium tuberculosis strain 1-0028P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041870.1 Mycobacterium tuberculosis strain 1-0023P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041871.1 Mycobacterium tuberculosis strain 1-0021P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041872.1 Mycobacterium tuberculosis strain 1-0017P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041873.1 Mycobacterium tuberculosis strain 1-0013P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041874.1 Mycobacterium tuberculosis strain 1-0009P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP041875.1 Mycobacterium tuberculosis strain 1-0007P6C4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP041876.1 Mycobacterium tuberculosis strain 1-0006P6C4 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP064405.1 Mycobacterium tuberculosis variant bovis BCG strain BCG SL 222 Sofia chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP070338.1 Mycobacterium tuberculosis strain 11502 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP071127.1 Mycobacterium tuberculosis strain 120/26CAO chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP071128.1 Mycobacterium tuberculosis strain 267/47W148 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP072762.1 Mycobacterium tuberculosis strain CG23 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP072764.1 Mycobacterium tuberculosis strain CG20 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP072763.1 Mycobacterium tuberculosis strain CG21 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP072761.1 Mycobacterium tuberculosis strain CG24 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP072765.1 Mycobacterium tuberculosis strain H37Rv_CG chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024671.1 Mycobacterium tuberculosis strain R2092 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089927.1 Mycobacterium tuberculosis strain M20 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP058236.1 Mycobacterium tuberculosis strain 060827 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP072790.1 Mycobacterium tuberculosis strain I0003447-5 chromosome, complete genome [96, 111, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069062.1 Mycobacterium tuberculosis strain N0069 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069063.1 Mycobacterium tuberculosis strain N3913 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1770]\n",
      "NZ_CP069064.1 Mycobacterium tuberculosis strain N1202 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069065.1 Mycobacterium tuberculosis strain N0091 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069066.1 Mycobacterium tuberculosis strain N1201 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069067.1 Mycobacterium tuberculosis strain N1268 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069068.1 Mycobacterium tuberculosis strain N1272 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069069.1 Mycobacterium tuberculosis strain N1176 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069070.1 Mycobacterium tuberculosis strain N1283 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069071.1 Mycobacterium tuberculosis strain N0136 chromosome, complete genome [96, 110, 197, 211, 1139, 1248, 1768]\n",
      "NZ_CP069072.1 Mycobacterium tuberculosis strain N1216 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069073.1 Mycobacterium tuberculosis strain N0054 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069074.1 Mycobacterium tuberculosis strain N1274 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069075.1 Mycobacterium tuberculosis strain N0004 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069076.1 Mycobacterium tuberculosis strain N0031 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069077.1 Mycobacterium tuberculosis strain N0072 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP069078.1 Mycobacterium tuberculosis strain N0157 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP095020.1 Mycobacterium tuberculosis strain SGF0232017 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP095021.1 Mycobacterium tuberculosis strain SGF0422017 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP095022.1 Mycobacterium tuberculosis strain SGF0472017 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP095023.1 Mycobacterium tuberculosis strain SGF0702019 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP110674.1 Mycobacterium tuberculosis strain BLR-31d chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP110619.1 Mycobacterium tuberculosis strain H37Rv-1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP112996.1 Mycobacterium tuberculosis variant bovis strain 2018/0565 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP112997.1 Mycobacterium tuberculosis variant bovis strain 2016/0386 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP115447.1 Mycobacterium tuberculosis strain BLR 9248 2019 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096839.1 Mycobacterium tuberculosis variant bovis strain Mb2487 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096840.1 Mycobacterium tuberculosis variant bovis strain Mb2269 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096841.1 Mycobacterium tuberculosis variant bovis strain Mb0820 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096842.1 Mycobacterium tuberculosis variant bovis strain Mb3114 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096843.1 Mycobacterium tuberculosis variant bovis strain Mb3602 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096844.1 Mycobacterium tuberculosis variant bovis strain Mb1855 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096845.1 Mycobacterium tuberculosis variant bovis strain Mb1101 chromosome, complete genome [96, 110, 198, 211, 646, 1122, 1139, 1248]\n",
      "NZ_CP096846.1 Mycobacterium tuberculosis variant bovis strain Mb2377 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096847.1 Mycobacterium tuberculosis variant bovis strain Mb0531 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP096848.1 Mycobacterium tuberculosis variant bovis strain Mb0486 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP085532.1 Mycobacterium tuberculosis variant bovis BCG strain Moreau PL chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP117298.1 Mycobacterium tuberculosis variant bovis strain OLF-AH-2022-TB-0122 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089611.1 Mycobacterium tuberculosis strain I0004240-3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089610.1 Mycobacterium tuberculosis strain I0004290-8 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089772.1 Mycobacterium tuberculosis strain I0003165-3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089773.1 Mycobacterium tuberculosis strain 01-R1430 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089774.1 Mycobacterium tuberculosis strain 02-R0894 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089775.1 Mycobacterium tuberculosis strain 02-R1708 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089776.1 Mycobacterium tuberculosis strain 02-R1896 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089777.1 Mycobacterium tuberculosis strain I0003270-1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089778.1 Mycobacterium tuberculosis strain I0003758-5 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089779.1 Mycobacterium tuberculosis strain 01-R1134 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089780.1 Mycobacterium tuberculosis strain M0003941-3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP089781.1 Mycobacterium tuberculosis strain 02-R1179 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP125620.1 Mycobacterium tuberculosis strain BLR 4299 2019 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP127275.1 Mycobacterium tuberculosis strain 6721 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP127273.1 Mycobacterium tuberculosis strain 3320 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP127274.1 Mycobacterium tuberculosis strain 17221 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP127276.1 Mycobacterium tuberculosis strain 5521 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP127277.1 Mycobacterium tuberculosis strain 25421 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130756.1 Mycobacterium tuberculosis strain MTb-Oman-3216462 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130758.1 Mycobacterium tuberculosis strain MTb-Oman-3215873 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130761.1 Mycobacterium tuberculosis strain MTb-Oman-3214927 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130763.1 Mycobacterium tuberculosis strain MTb-Oman-3214805 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130767.1 Mycobacterium tuberculosis strain MTb-Oman-3214642 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130774.1 Mycobacterium tuberculosis strain MTb-Oman-3214168 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130784.1 Mycobacterium tuberculosis strain MTb-Oman-3213509 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130802.1 Mycobacterium tuberculosis strain MTb-Oman-3212387 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130811.1 Mycobacterium tuberculosis strain MTb-Oman-321857 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP130817.1 Mycobacterium tuberculosis strain MTb-Oman-32193 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP133041.1 Mycobacterium tuberculosis strain AST-T3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP133042.1 Mycobacterium tuberculosis strain AST-T4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP133043.1 Mycobacterium tuberculosis strain AST-T5 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP133044.1 Mycobacterium tuberculosis strain AST-T6 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP133045.1 Mycobacterium tuberculosis strain AST-T7 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP097109.1 Mycobacterium tuberculosis strain MTBCR170031 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP097110.1 Mycobacterium tuberculosis strain MTBCR170027 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP097111.1 Mycobacterium tuberculosis strain MTBCR170020 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP097112.1 Mycobacterium tuberculosis strain MTBCR170001 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024454.1 Mycobacterium tuberculosis strain DN049 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024455.1 Mycobacterium tuberculosis strain DN059 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024456.1 Mycobacterium tuberculosis strain DN067 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024457.1 Mycobacterium tuberculosis strain DN068 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024458.1 Mycobacterium tuberculosis strain DN101 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024459.1 Mycobacterium tuberculosis strain DN105 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024460.1 Mycobacterium tuberculosis strain DN146 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024462.1 Mycobacterium tuberculosis strain DN251 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_AP024461.1 Mycobacterium tuberculosis strain DN181 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP149780.1 Mycobacterium tuberculosis strain ITM-1999-01508 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP149781.1 Mycobacterium tuberculosis strain ITM-2000-00448 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP150983.2 Mycobacterium tuberculosis strain M64 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP152406.1 Mycobacterium tuberculosis strain MTBCR170913 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP154611.1 Mycobacterium tuberculosis strain ITM-2005-02460 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP154613.1 Mycobacterium tuberculosis strain ITM-2021-02250 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP154612.1 Mycobacterium tuberculosis strain ITM-2015-00021 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP136197.1 Mycobacterium tuberculosis strain 1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP136199.1 Mycobacterium tuberculosis strain 3 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP136202.1 Mycobacterium tuberculosis strain 6 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP136204.1 Mycobacterium tuberculosis strain 8 chromosome, complete genome [96, 110, 198, 211, 1248, 2907]\n",
      "NZ_CP136205.1 Mycobacterium tuberculosis strain 9 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157900.1 Mycobacterium tuberculosis strain H-19-0007 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157901.1 Mycobacterium tuberculosis strain H-19-0008 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157902.1 Mycobacterium tuberculosis strain H-19-0027 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157903.1 Mycobacterium tuberculosis strain H-20-0022 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157904.1 Mycobacterium tuberculosis strain H-20-0024 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157905.1 Mycobacterium tuberculosis strain H-21-0019 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP159543.1 Mycobacterium tuberculosis strain CA02 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP158171.1 Mycobacterium tuberculosis strain 15 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP157911.1 Mycobacterium tuberculosis strain 15 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP143267.1 Mycobacterium tuberculosis strain PSNK363 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP160832.1 Mycobacterium tuberculosis variant bovis strain 2023/0113 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP172229.1 Mycobacterium tuberculosis str. Erdman = ATCC 35801 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP133040.1 Mycobacterium tuberculosis strain AST-T2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP149484.1 Mycobacterium tuberculosis strain mc26030 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP150074.1 Mycobacterium tuberculosis strain ITM-2004-01893 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP150095.1 Mycobacterium tuberculosis strain ITM-1992-08557 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP154610.1 Mycobacterium tuberculosis strain ITM-2002-00926 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP154614.1 Mycobacterium tuberculosis subsp. tuberculosis strain ITM-1968-01344 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP154615.1 Mycobacterium tuberculosis subsp. tuberculosis strain ITM-2003-01539 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP159623.1 Mycobacterium tuberculosis strain CA03 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP159515.1 Mycobacterium tuberculosis strain CB03 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP159542.1 Mycobacterium tuberculosis strain CA01 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP159786.1 Mycobacterium tuberculosis strain CB01 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP159622.1 Mycobacterium tuberculosis strain CB02 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP178838.1 Mycobacterium tuberculosis strain A16 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP188097.1 Mycobacterium tuberculosis str. Erdman = ATCC 35801 strain Erdman chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP189862.1 Mycobacterium tuberculosis variant caprae strain SY-1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP119412.1 Mycobacterium tuberculosis strain CDC1551 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194235.1 Mycobacterium tuberculosis strain LP-0504069-RM2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194236.1 Mycobacterium tuberculosis strain LP-0504061-RM2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194237.1 Mycobacterium tuberculosis strain LP-0497747-RM1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194238.1 Mycobacterium tuberculosis strain LP-0474076-RM3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194239.1 Mycobacterium tuberculosis strain LP-0474076-RM2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194240.1 Mycobacterium tuberculosis strain LP-0474076-RM1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194241.1 Mycobacterium tuberculosis strain LP-0333226-RM3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194242.1 Mycobacterium tuberculosis strain LP-0333226-RM2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194243.1 Mycobacterium tuberculosis strain LP-0261867-RM7 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194245.1 Mycobacterium tuberculosis strain LP-0261867-RM1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194244.1 Mycobacterium tuberculosis strain LP-0261867-RM5 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194247.1 Mycobacterium tuberculosis strain LP-0257184-RM5 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194246.1 Mycobacterium tuberculosis strain LP-0257184-RM6 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194248.1 Mycobacterium tuberculosis strain LP-0257184-RM4 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194249.1 Mycobacterium tuberculosis strain LP-0257184-RM3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194251.1 Mycobacterium tuberculosis strain LP-0107828-RM5 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194250.1 Mycobacterium tuberculosis strain LP-0107828-RM6 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194252.1 Mycobacterium tuberculosis strain LP-0107828-RM1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194253.1 Mycobacterium tuberculosis strain LP-0106967-RM5 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194254.1 Mycobacterium tuberculosis strain LP-0106963-RM3 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194255.1 Mycobacterium tuberculosis strain LP-0106963-RM2 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_CP194256.1 Mycobacterium tuberculosis strain LP-0106963-RM1 chromosome, complete genome [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR027516.1 Mycobacterium tuberculosis strain DKC2 isolate Mycobacterium tuberculosis chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR699570.1 Mycobacterium tuberculosis variant bovis strain Mb3601 isolate 14Z005608 chromosome Mb3601 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR882499.1 Mycobacterium tuberculosis variant microti OV254 isolate vole chromosome MmicOV254 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR882496.1 Mycobacterium tuberculosis variant microti strain ATCC 35782 isolate vole chromosome MmicATCC35782 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR882498.1 Mycobacterium tuberculosis variant microti strain Maus III isolate human chromosome MmicMaus3 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR882497.1 Mycobacterium tuberculosis variant microti strain Maus IV isolate human chromosome MmicMaus4 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_LR882500.1 Mycobacterium tuberculosis variant microti strain 94-2272 isolate human chromosome Mmic942272 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OW052189.1 Mycobacterium tuberculosis strain Lineage 2.2.7 isolate Mycobacterium tuberculosis chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OW052302.1 Mycobacterium tuberculosis strain Lineage 3.1.1 isolate Mycobacterium tuberculosis chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OW052571.1 Mycobacterium tuberculosis strain Lineage 4.1 isolate Mycobacterium tuberculosis chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OW052570.1 Mycobacterium tuberculosis strain Lineage 1.1.2 isolate Mycobacterium tuberculosis chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OY741334.1 Mycobacterium tuberculosis isolate GReAT_CRyPTIC_Unmapped_H37Rv chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023108.1 Mycobacterium tuberculosis strain P010-N1385 isolate P010 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023109.1 Mycobacterium tuberculosis strain N1015 isolate N1015 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023110.1 Mycobacterium tuberculosis strain P001-N1377 isolate N1377 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023111.1 Mycobacterium tuberculosis strain P028-N1362 isolate N1362 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023113.1 Mycobacterium tuberculosis strain P007-N1108 isolate P007 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023112.1 Mycobacterium tuberculosis strain P074-N1402 isolate N1402 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023114.1 Mycobacterium tuberculosis strain P008-N1380 isolate P008 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023115.1 Mycobacterium tuberculosis strain P006-N1591 isolate P006 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023117.1 Mycobacterium tuberculosis strain P020-N1386 isolate N1386 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023116.1 Mycobacterium tuberculosis strain P003-N1374 isolate N1374 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023118.1 Mycobacterium tuberculosis strain P034-N1426 isolate N1426 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023119.1 Mycobacterium tuberculosis strain P022-N1431 isolate N1431 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023120.1 Mycobacterium tuberculosis strain P059-N1392 isolate N1392 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023121.1 Mycobacterium tuberculosis strain P052-N1429 isolate N1429 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023122.1 Mycobacterium tuberculosis strain P073-N1394 isolate N1394 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023123.1 Mycobacterium tuberculosis strain P042-N1430 isolate N1430 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ023124.1 Mycobacterium tuberculosis strain P066-N1411 isolate N1411 chromosome 1, complete sequence [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222402.1 Mycobacterium tuberculosis isolate WBB1456_11-00225-4 chromosome WBB1456 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222403.1 Mycobacterium tuberculosis isolate WBB1454_IB091-1 chromosome WBB1454 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222404.1 Mycobacterium tuberculosis isolate WBB1453_11-00429-1 chromosome WBB1453 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222435.1 Mycobacterium tuberculosis isolate WBB1009_SL1875 chromosome WBB1009 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222436.1 Mycobacterium tuberculosis isolate WBB1008_SL1975 chromosome WBB1008 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222505.1 Mycobacterium tuberculosis isolate WBB445_ARS7496 chromosome WBB445 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222514.1 Mycobacterium tuberculosis isolate WBB447_G67 chromosome WBB447 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222538.1 Mycobacterium tuberculosis isolate WBB1007_LQ1975 chromosome WBB1007 [96, 110, 198, 211, 1139, 1248, 1768]\n",
      "NZ_OZ222539.1 Mycobacterium tuberculosis isolate WBB1452_10-01964-2 chromosome WBB1452 [96, 110, 198, 211, 1139, 1248, 1768]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import  SeqIO\n",
    "from Bio.Restriction import *\n",
    "\n",
    "r=RestrictionBatch()\n",
    "path = \"F:/TEST/tb/ncbi_dataset/data\"\n",
    "import os\n",
    "\n",
    "\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "P1=\"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P=Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2=str(P)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(str(path) + \"/\" + str(i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(str(path) + \"/\" + str(i) + \"/\" + str(s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                if P1 in SEQUENCE:\n",
    "                    if P2 in SEQUENCE:\n",
    "                        subsequence = SEQUENCE[SEQUENCE.index(str(P1)):SEQUENCE.index(str(P2)) + len( str(P2))]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                            if len(g) < 2:\n",
    "                                break\n",
    "                            for f in g:\n",
    "                                if len(f) in sorttt:\n",
    "                                    continue\n",
    "                                if len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            print(desc, sorttt)\n",
    "                        except:\n",
    "                            pass\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())\n",
    "                    m = m[::-1]\n",
    "                    if P1 in m:\n",
    "                        if P2 in m:\n",
    "                            assemply_sequence = m[m.index(str(P1)):m.index(str(P2)) + len(str(P2))]\n",
    "                            try:\n",
    "                                sorttt = []\n",
    "                                g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                                if len(g) < 2:\n",
    "                                    break\n",
    "                                for f in g:\n",
    "                                    if len(f) in sorttt:\n",
    "                                        continue\n",
    "                                    if len(f) >= 50:\n",
    "                                        sorttt.append(len(f))\n",
    "                                sorttt.sort()\n",
    "                                print(desc, sorttt)\n",
    "                            except:\n",
    "                                pass\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "388d5295-304d-4d13-a2da-c858b57b3e1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_015758.1 Mycobacterium tuberculosis variant africanum GM041182, complete sequence [96, 110, 200, 331, 940, 1545, 1768]\n",
      "NZ_CP014617.1 Mycobacterium tuberculosis variant africanum strain UT307 chromosome [96, 110, 200, 331, 940, 1545, 1768]\n",
      "NZ_LR993210.1 Mycobacterium tuberculosis variant africanum isolate Various chromosome MAF01, whole genome shotgun sequence [96, 110, 200, 331, 940, 1545, 1768]\n",
      "NZ_LR993208.1 Mycobacterium tuberculosis variant africanum isolate Various chromosome MAF65, whole genome shotgun sequence [96, 110, 200, 331, 940, 1545, 1768]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import  SeqIO\n",
    "from Bio.Restriction import *\n",
    "\n",
    "r=RestrictionBatch()\n",
    "path = \"F:/dr ibrhim/ncbi/africanum/ncbi_dataset/data\"\n",
    "import os\n",
    "\n",
    "\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "P1=\"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P=Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2=str(P)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(str(path) + \"/\" + str(i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(str(path) + \"/\" + str(i) + \"/\" + str(s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                if P1 in SEQUENCE:\n",
    "                    if P2 in SEQUENCE:\n",
    "                        subsequence = SEQUENCE[SEQUENCE.index(str(P1)):SEQUENCE.index(str(P2)) + len( str(P2))]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                            if len(g) < 2:\n",
    "                                break\n",
    "                            for f in g:\n",
    "                                if len(f) in sorttt:\n",
    "                                    continue\n",
    "                                if len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            print(desc, sorttt)\n",
    "                        except:\n",
    "                            pass\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())\n",
    "                    m = m[::-1]\n",
    "                    if P1 in m:\n",
    "                        if P2 in m:\n",
    "                            assemply_sequence = m[m.index(str(P1)):m.index(str(P2)) + len(str(P2))]\n",
    "                            try:\n",
    "                                sorttt = []\n",
    "                                g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                                if len(g) < 2:\n",
    "                                    break\n",
    "                                for f in g:\n",
    "                                    if len(f) in sorttt:\n",
    "                                        continue\n",
    "                                    if len(f) >= 50:\n",
    "                                        sorttt.append(len(f))\n",
    "                                sorttt.sort()\n",
    "                                print(desc, sorttt)\n",
    "                            except:\n",
    "                                pass\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1db8c982-498b-4c03-b7e4-404998a758e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                           Description             Fragment_Length\n",
      "NZ_LR130759.1 NZ_LR130759.1 Mycobacterium basiliense isolate 901379 chromosome I, complete sequence [308, 452, 892, 1347, 1742]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/basiliense/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6a1eb98a-4a09-45d4-bb63-563c8156bee7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                                                                               Description                       Fragment_Length\n",
      "  NC_008769.1                                                         NC_008769.1 1 Mycobacterium tuberculosis variant bovis BCG str. Pasteur 1173P2, complete sequence [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "  NC_012207.1                                                              NC_012207.1 1 Mycobacterium tuberculosis variant bovis BCG str. Tokyo 172, complete sequence [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP003494.1                                      NZ_CP003494.1 1 Mycobacterium tuberculosis variant bovis BCG str. ATCC 35743 strain Tice chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "  NC_002945.4                               NC_002945.4 4 Mycobacterium tuberculosis variant bovis AF2122/97 chromosome Mycobacterium_bovis_AF212297, complete sequence [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "  NC_016804.1                                                                 NC_016804.1 1 Mycobacterium tuberculosis variant bovis BCG str. Mexico, complete sequence [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "  NC_020245.2                                                            NC_020245.2 2 Mycobacterium tuberculosis variant bovis BCG str. Korea 1168P, complete sequence [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP015773.2                                                          NZ_CP015773.2 2 Mycobacterium tuberculosis variant bovis strain SP38 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_AM412059.1 NZ_AM412059.1 1 Mycobacterium tuberculosis variant bovis BCG str. Moreau RDJ isolate SL21 FAP RJ Passage B8S2 vaccine culture chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP008744.1                                                      NZ_CP008744.1 1 Mycobacterium tuberculosis variant bovis BCG strain 3281 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP012095.1                                                          NZ_CP012095.1 1 Mycobacterium tuberculosis variant bovis strain 1595 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP009243.1                                                NZ_CP009243.1 1 Mycobacterium tuberculosis variant bovis BCG strain Russia 368 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP011455.1                                                         NZ_CP011455.1 1 Mycobacterium tuberculosis variant bovis strain BCG-1 isolate 368 shch chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP013741.1                                                NZ_CP013741.1 1 Mycobacterium tuberculosis variant bovis strain BCG-1 (Russia) chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP014566.1                                                   NZ_CP014566.1 1 Mycobacterium tuberculosis variant bovis BCG str. Tokyo 172 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP023708.2                                                                         NZ_CP023708.2 2 Mycobacterium tuberculosis variant bovis strain 3/86Rv chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP027035.1                                                     NZ_CP027035.1 1 Mycobacterium tuberculosis variant bovis strain 2002/0476 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP033310.1                                                                    NZ_CP033310.1 1 Mycobacterium tuberculosis variant bovis BCG strain BCG-S48 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP033311.1                                                                    NZ_CP033311.1 1 Mycobacterium tuberculosis variant bovis BCG strain BCG_S49 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP039851.1                                             NZ_CP039851.1 1 Mycobacterium tuberculosis variant bovis strain Danish delta-sapM chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP039850.1                                                   NZ_CP039850.1 1 Mycobacterium tuberculosis variant bovis strain Danish 1331 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP040832.1                                                                              NZ_CP040832.1 1 Mycobacterium tuberculosis variant bovis strain 1 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP064405.1                                          NZ_CP064405.1 1 Mycobacterium tuberculosis variant bovis BCG strain BCG SL 222 Sofia chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP109681.1                                                         NZ_CP109681.1 1 Mycobacterium tuberculosis variant bovis strain BCG Pasteur ATCC 35734 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP110223.1                                                      NZ_CP110223.1 1 Mycobacterium tuberculosis variant bovis strain BCG delta BCG1419c mutant chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP112996.1                                                     NZ_CP112996.1 1 Mycobacterium tuberculosis variant bovis strain 2018/0565 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP112997.1                                                     NZ_CP112997.1 1 Mycobacterium tuberculosis variant bovis strain 2016/0386 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096839.1                                                        NZ_CP096839.1 1 Mycobacterium tuberculosis variant bovis strain Mb2487 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096840.1                                                        NZ_CP096840.1 1 Mycobacterium tuberculosis variant bovis strain Mb2269 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096841.1                                                        NZ_CP096841.1 1 Mycobacterium tuberculosis variant bovis strain Mb0820 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096842.1                                                        NZ_CP096842.1 1 Mycobacterium tuberculosis variant bovis strain Mb3114 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096843.1                                                        NZ_CP096843.1 1 Mycobacterium tuberculosis variant bovis strain Mb3602 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096844.1                                                        NZ_CP096844.1 1 Mycobacterium tuberculosis variant bovis strain Mb1855 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096845.1                                                        NZ_CP096845.1 1 Mycobacterium tuberculosis variant bovis strain Mb1101 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096846.1                                                        NZ_CP096846.1 1 Mycobacterium tuberculosis variant bovis strain Mb2377 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096847.1                                                        NZ_CP096847.1 1 Mycobacterium tuberculosis variant bovis strain Mb0531 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP096848.1                                                        NZ_CP096848.1 1 Mycobacterium tuberculosis variant bovis strain Mb0486 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP085532.1                                                 NZ_CP085532.1 1 Mycobacterium tuberculosis variant bovis BCG strain Moreau PL chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP117298.1                                           NZ_CP117298.1 1 Mycobacterium tuberculosis variant bovis strain OLF-AH-2022-TB-0122 chromosome, complete genome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP133602.1                                                                     NZ_CP133602.1 1 Mycobacterium tuberculosis variant bovis isolate PHOPR KO2 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP133604.1                                                                      NZ_CP133604.1 1 Mycobacterium tuberculosis variant bovis strain AF2122/97 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_CP133601.1                                                                     NZ_CP133601.1 1 Mycobacterium tuberculosis variant bovis isolate PHOPR KO3 chromosome [96, 110, 198, 232, 1139, 1520, 1700]\n",
      "NZ_LR699570.1                                                NZ_LR699570.1 1 Mycobacterium tuberculosis variant bovis strain Mb3601 isolate 14Z005608 chromosome Mb3601 [96, 110, 198, 232, 1139, 1520, 1700]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/bovis\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a26e585b-0b80-4751-a9a9-ec2d96df64b8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                                              Description                       Fragment_Length\n",
      "  NC_015848.1                                                     NC_015848.1 Mycobacterium canettii CIPT 140010059, complete sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "  NC_019951.1                                                     NC_019951.1 Mycobacterium canettii CIPT 140070010, complete sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "  NC_019950.1                                                     NC_019950.1 Mycobacterium canettii CIPT 140060008, complete sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "  NC_019952.1                                                     NC_019952.1 Mycobacterium canettii CIPT 140070017, complete sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "  NC_019965.1                                                     NC_019965.1 Mycobacterium canettii CIPT 140070008, complete sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_CP141297.1                                            NZ_CP141297.1 Mycobacterium canetti strain ET1291 chromosome, complete genome [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_LR993209.1 NZ_LR993209.1 Mycobacterium canettii CIPT 140070007 isolate STB-I chromosome CAOO01000000, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_LR993206.1 NZ_LR993206.1 Mycobacterium canettii CIPT 140070002 isolate STB-E chromosome CAOL01000000, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_LR993207.2 NZ_LR993207.2 Mycobacterium canettii CIPT 140070005 isolate STB-G chromosome CAOM01000000, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU916521.1            NZ_OU916521.1 Mycobacterium canetti isolate Percy1078 chromosome McanPercy1078, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU916613.1              NZ_OU916613.1 Mycobacterium canetti isolate Percy975 chromosome McanPercy975, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU916928.1            NZ_OU916928.1 Mycobacterium canetti isolate Percy1129 chromosome McanPercy1129, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU916936.1              NZ_OU916936.1 Mycobacterium canetti isolate Percy525 chromosome McanPercy525, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917094.1            NZ_OU917094.1 Mycobacterium canetti isolate Percy1004 chromosome McanPercy1004, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917211.1            NZ_OU917211.1 Mycobacterium canetti isolate Percy1102 chromosome McanPercy1102, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917557.1                NZ_OU917557.1 Mycobacterium canetti isolate Percy79 chromosome McanPercy79, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917796.1            NZ_OU917796.1 Mycobacterium canetti isolate Percy1084 chromosome McanPercy1084, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917819.1            NZ_OU917819.1 Mycobacterium canetti isolate Percy1130 chromosome McanPercy1130, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917987.1            NZ_OU917987.1 Mycobacterium canetti isolate Percy1101 chromosome McanPercy1101, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917921.1              NZ_OU917921.1 Mycobacterium canetti isolate Percy258 chromosome McanPercy258, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917923.1                NZ_OU917923.1 Mycobacterium canetti isolate Percy50 chromosome McanPercy50, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917922.1                NZ_OU917922.1 Mycobacterium canetti isolate Percy22 chromosome McanPercy22, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917986.1              NZ_OU917986.1 Mycobacterium canetti isolate Percy301 chromosome McanPercy301, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917918.1            NZ_OU917918.1 Mycobacterium canetti isolate Percy1116 chromosome McanPercy1116, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917919.1            NZ_OU917919.1 Mycobacterium canetti isolate Percy1115 chromosome McanPercy1115, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917920.1                NZ_OU917920.1 Mycobacterium canetti isolate Percy32 chromosome McanPercy32, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OU917917.1            NZ_OU917917.1 Mycobacterium canetti isolate Percy1105 chromosome McanPercy1105, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n",
      "NZ_OX291665.1 NZ_OX291665.1 Mycobacterium canettii CIPT 140070013 isolate STB-H chromosome CAON01000000, whole genome shotgun sequence [96, 110, 198, 232, 1139, 1346, 1840]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/canetti/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "74c0ba57-3fca-4337-81f5-c9b25ce0a5af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                        Description             Fragment_Length\n",
      "NZ_AP022613.1 NZ_AP022613.1 Mycobacterium conspicuum strain JCM 14738 chromosome [95, 209, 1139, 1248, 2039]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/conspicuum/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ec0d9c9e-6afc-4073-a891-9af17a3ee7fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                     Description                  Fragment_Length\n",
      "NZ_AP022569.1 NZ_AP022569.1 Mycobacterium cookii strain JCM 12404 chromosome, complete genome [109, 423, 592, 928, 1346, 1379]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/cookii/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "cde87779-1950-48cb-8fa9-a48f76320148",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                          Description             Fragment_Length\n",
      "NZ_AP022576.1 NZ_AP022576.1 Mycobacterium florentinum strain JCM 14740 chromosome, complete genome [306, 408, 929, 1340, 1726]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/florentium/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "82c843c3-ee9e-4f41-89db-0a9f5ec2396b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                          Description             Fragment_Length\n",
      "NZ_AP022576.1 NZ_AP022576.1 Mycobacterium florentinum strain JCM 14740 chromosome, complete genome [306, 408, 929, 1340, 1726]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/florentium/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "aa156715-f864-4b8e-9080-160a482d32be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                          Description                 Fragment_Length\n",
      "NZ_CP076381.1 NZ_CP076381.1 Mycobacterium intracellulare strain S1-36A chromosome, complete genome [308, 452, 854, 889, 893, 1349]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/intracellulare/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "41b3d6e3-1667-4536-8d13-715b79e9e14e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                          Description             Fragment_Length\n",
      "NZ_AP022576.1 NZ_AP022576.1 Mycobacterium florentinum strain JCM 14740 chromosome, complete genome [306, 408, 929, 1340, 1726]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/florentium/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a8c477f0-add5-40a8-ae64-78f5f7a608c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                           Description                            Fragment_Length\n",
      "  NC_022663.1                      NC_022663.1 Mycobacterium kansasii ATCC 12478, complete sequence [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP019883.1                            NZ_CP019883.1 Mycobacterium kansasii strain 1MK chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP019884.1                            NZ_CP019884.1 Mycobacterium kansasii strain 4MK chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP019885.1                            NZ_CP019885.1 Mycobacterium kansasii strain 6MK chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP019886.1                           NZ_CP019886.1 Mycobacterium kansasii strain 10MK chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP019887.1                           NZ_CP019887.1 Mycobacterium kansasii strain 11MK chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP019888.1                            NZ_CP019888.1 Mycobacterium kansasii strain 9MK chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP084364.1                  NZ_CP084364.1 Mycobacterium kansasii strain FDAARGOS_1534 chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP089216.1 NZ_CP089216.1 Mycobacterium kansasii strain FDAARGOS_1615 chromosome, complete genome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP089218.1 NZ_CP089218.1 Mycobacterium kansasii strain FDAARGOS_1614 chromosome, complete genome [123, 164, 225, 360, 373, 798, 1063, 1631]\n",
      "NZ_CP089231.1                  NZ_CP089231.1 Mycobacterium kansasii strain FDAARGOS_1616 chromosome [123, 164, 225, 360, 373, 798, 1063, 1631]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/kansasii/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = BsaWI.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = BsaWI.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "e799deb4-dfef-47c4-929a-8e1f51487ced",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                               Description                   Fragment_Length\n",
      "NZ_AP022581.1                            NZ_AP022581.1 Mycobacterium lacus strain JCM 15657 chromosome, complete genome [209, 308, 430, 1140, 1338, 1344]\n",
      "NZ_OY970455.1 NZ_OY970455.1 Mycobacterium lacus strain DSM 44577 isolate human chromosome Mlac_44577, complete sequence [209, 308, 430, 1140, 1338, 1344]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/lacus/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "80ae9f4f-596c-49d2-bb7f-93094baae885",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                       Description              Fragment_Length\n",
      "  NC_011896.1                 NC_011896.1 Mycobacterium leprae Br4923, complete genome sequence [223, 562, 1140, 1344, 1508]\n",
      "  NC_002677.1                            NC_002677.1 Mycobacterium leprae TN, complete sequence [223, 562, 1140, 1344, 1508]\n",
      "NZ_CP029543.1 NZ_CP029543.1 Mycobacterium leprae strain MRHRU-235-G chromosome, complete genome [223, 562, 1140, 1344, 1508]\n",
      "NZ_AP014567.1            NZ_AP014567.1 Mycobacterium leprae Kyoto-2 chromosome, complete genome [223, 562, 1140, 1344, 1508]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/leprae/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "992d213c-64aa-43ca-91ff-2c9bde9d9463",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                       Description         Fragment_Length\n",
      "NZ_AP022590.1 NZ_AP022590.1 Mycobacterium mantenii strain JCM 18113 chromosome, complete genome [308, 1341, 1346, 1750]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/mantenii/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "6a1f0c78-e107-4670-8629-4eb50b436c5e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                                      Description                       Fragment_Length\n",
      "NZ_LR882499.1                 NZ_LR882499.1 Mycobacterium tuberculosis variant microti OV254 isolate vole chromosome MmicOV254 [95, 110, 198, 311, 1190, 1400, 1740]\n",
      "NZ_LR882496.1 NZ_LR882496.1 Mycobacterium tuberculosis variant microti strain ATCC 35782 isolate vole chromosome MmicATCC35782 [95, 110, 198, 311, 1190, 1400, 1740]\n",
      "NZ_LR882498.1      NZ_LR882498.1 Mycobacterium tuberculosis variant microti strain Maus III isolate human chromosome MmicMaus3 [95, 110, 198, 311, 1190, 1400, 1740]\n",
      "NZ_LR882497.1       NZ_LR882497.1 Mycobacterium tuberculosis variant microti strain Maus IV isolate human chromosome MmicMaus4 [95, 110, 198, 311, 1190, 1400, 1740]\n",
      "NZ_LR882500.1      NZ_LR882500.1 Mycobacterium tuberculosis variant microti strain 94-2272 isolate human chromosome Mmic942272 [95, 110, 198, 311, 1190, 1400, 1740]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/micrtio\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "4726ac9e-2897-4ef1-9a6d-d96068725589",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                                            Description              Fragment_Length\n",
      "  NC_002944.2                                        NC_002944.2 Mycobacterium avium subsp. paratuberculosis K-10, complete sequence [228, 308, 1140, 1627, 1749]\n",
      "  NC_008595.1                                                                 NC_008595.1 Mycobacterium avium 104, complete sequence [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP053749.1                             NZ_CP053749.1 Mycobacterium avium subsp. paratuberculosis S397 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "  NC_021200.1                                        NC_021200.1 Mycobacterium avium subsp. paratuberculosis MAP4, complete sequence [228, 308, 1140, 1627, 1749]\n",
      "NZ_AP012555.1                                 NZ_AP012555.1 Mycobacterium avium subsp. hominissuis TH135 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP010113.1                        NZ_CP010113.1 Mycobacterium avium subsp. paratuberculosis strain E1 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP010114.1                       NZ_CP010114.1 Mycobacterium avium subsp. paratuberculosis strain E93 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP015495.1 NZ_CP015495.1 Mycobacterium avium subsp. paratuberculosis strain MAP/TANUVAS/TN/India/2008 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP016396.1                         NZ_CP016396.1 Mycobacterium avium strain RCAD0278 isolate SC150824 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP009360.4                         NZ_CP009360.4 Mycobacterium avium subsp. hominissuis strain OCU464 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP018363.1                            NZ_CP018363.1 Mycobacterium avium subsp. hominissuis strain H87 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP022095.2              NZ_CP022095.2 Mycobacterium avium subsp. paratuberculosis strain FDAARGOS_305 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP022105.1                                   NZ_CP022105.1 Mycobacterium avium subsp. paratuberculosis strain JII-1961 chromosome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP076851.1                         NZ_CP076851.1 Mycobacterium avium subsp. hominissuis strain CAM177 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP016818.3                           NZ_CP016818.3 Mycobacterium avium subsp. hominissuis strain HP17 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP018014.3                  NZ_CP018014.3 Mycobacterium avium subsp. hominissuis strain OCU901s_S2_2s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP018020.3                  NZ_CP018020.3 Mycobacterium avium subsp. hominissuis strain OCU873s_P7_4s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP029332.1                         NZ_CP029332.1 Mycobacterium avium subsp. hominissuis strain MAC109 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP028731.1                                               NZ_CP028731.1 Mycobacterium avium strain HJW chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP033428.1               NZ_CP033428.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_CN7/15 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP033427.1               NZ_CP033427.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_CN9/15 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP033910.1               NZ_CP033910.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_CN4/13 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP033911.1              NZ_CP033911.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_JB16/15 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP033909.1               NZ_CP033909.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_JJ1/13 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP033688.1                   NZ_CP033688.1 Mycobacterium avium subsp. paratuberculosis strain Telford chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP036220.2                       NZ_CP036220.2 Mycobacterium avium subsp. hominissuis strain mc2 2500 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP040250.1                         NZ_CP040250.1 Mycobacterium avium subsp. hominissuis strain 101174 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP040247.1                         NZ_CP040247.1 Mycobacterium avium subsp. hominissuis strain 101034 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP040255.1                         NZ_CP040255.1 Mycobacterium avium subsp. hominissuis strain 101115 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_AP020326.1                         NZ_AP020326.1 Mycobacterium avium subsp. hominissuis strain JP-H-1 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP046507.1                            NZ_CP046507.1 Mycobacterium avium subsp. avium strain DSM 44156 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP053068.1                 NZ_CP053068.1 Mycobacterium avium subsp. paratuberculosis strain DSM 44135 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP066812.1                   NZ_CP066812.1 Mycobacterium avium subsp. paratuberculosis strain 42-13-1 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP060405.1                                              NZ_CP060405.1 Mycobacterium avium subsp. hominissuis strain W9 chromosome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP060407.1                                             NZ_CP060407.1 Mycobacterium avium subsp. hominissuis strain w14 chromosome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP085977.1                        NZ_CP085977.1 Mycobacterium avium subsp. avium strain FDAARGOS_1607 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP085978.1                        NZ_CP085978.1 Mycobacterium avium subsp. avium strain FDAARGOS_1606 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP089223.1                        NZ_CP089223.1 Mycobacterium avium subsp. avium strain FDAARGOS_1609 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP089230.1                                         NZ_CP089230.1 Mycobacterium avium subsp. avium strain FDAARGOS_1608 chromosome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP018019.1                 NZ_CP018019.1 Mycobacterium avium subsp. hominissuis strain OCU889s_P11_4s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP106873.1                             NZ_CP106873.1 Mycobacterium avium subsp. paratuberculosis K-10 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP113485.1                      NZ_CP113485.1 Mycobacterium avium subsp. paratuberculosis strain LN20 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP139433.1                      NZ_CP139433.1 Mycobacterium avium subsp. paratuberculosis strain 6756 chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP083993.1                  NZ_CP083993.1 Mycobacterium avium subsp. hominissuis strain OCU855s_P2_4s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP083998.1                  NZ_CP083998.1 Mycobacterium avium subsp. hominissuis strain OCU898s_S1_3s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP085194.1                   NZ_CP085194.1 Mycobacterium avium subsp. hominissuis strain OCU863_P4-4s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n",
      "NZ_CP085280.1                   NZ_CP085280.1 Mycobacterium avium subsp. hominissuis strain OCU859_P3-3s chromosome, complete genome [228, 308, 1140, 1627, 1749]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/ncbi_avium/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c2207853-45a4-40ae-9cb0-d3eda498f631",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             Name                                                                                     Description         Fragment_Length\n",
      "    NZ_CP011883.2 NZ_CP011883.2 Mycobacterium haemophilum DSM 44634 strain ATCC 29548 chromosome, complete genome [210, 1140, 1344, 2079]\n",
      "NZ_LDPR01000004.1  NZ_LDPR01000004.1 Mycobacterium haemophilum strain UC3 contig_4, whole genome shotgun sequence [210, 1140, 1344, 2079]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/haemophilum/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "900fab6f-2e34-4891-b043-dff9102aecb0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                         Description                 Fragment_Length\n",
      "NZ_CP092488.2 NZ_CP092488.2 Mycobacterium paraterrae strain DSM 45127 chromosome, complete genome [79, 422, 596, 928, 1268, 1488]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/paraterrae/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c504fb57-74ff-43ca-8e47-37c292a2689c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                                            Description              Fragment_Length\n",
      "  NC_002944.2                                        NC_002944.2 Mycobacterium avium subsp. paratuberculosis K-10, complete sequence [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP053749.1                             NZ_CP053749.1 Mycobacterium avium subsp. paratuberculosis S397 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "  NC_021200.1                                        NC_021200.1 Mycobacterium avium subsp. paratuberculosis MAP4, complete sequence [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP010113.1                        NZ_CP010113.1 Mycobacterium avium subsp. paratuberculosis strain E1 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP010114.1                       NZ_CP010114.1 Mycobacterium avium subsp. paratuberculosis strain E93 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP015495.1 NZ_CP015495.1 Mycobacterium avium subsp. paratuberculosis strain MAP/TANUVAS/TN/India/2008 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP022095.2              NZ_CP022095.2 Mycobacterium avium subsp. paratuberculosis strain FDAARGOS_305 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP022105.1                                   NZ_CP022105.1 Mycobacterium avium subsp. paratuberculosis strain JII-1961 chromosome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP033428.1               NZ_CP033428.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_CN7/15 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP033427.1               NZ_CP033427.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_CN9/15 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP033910.1               NZ_CP033910.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_CN4/13 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP033909.1               NZ_CP033909.1 Mycobacterium avium subsp. paratuberculosis strain MAPK_JJ1/13 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP033688.1                   NZ_CP033688.1 Mycobacterium avium subsp. paratuberculosis strain Telford chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP053068.1                 NZ_CP053068.1 Mycobacterium avium subsp. paratuberculosis strain DSM 44135 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP066812.1                   NZ_CP066812.1 Mycobacterium avium subsp. paratuberculosis strain 42-13-1 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP106873.1                             NZ_CP106873.1 Mycobacterium avium subsp. paratuberculosis K-10 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP113485.1                      NZ_CP113485.1 Mycobacterium avium subsp. paratuberculosis strain LN20 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n",
      "NZ_CP139433.1                      NZ_CP139433.1 Mycobacterium avium subsp. paratuberculosis strain 6756 chromosome, complete genome [129, 350, 1140, 1429, 1748]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/paratuberculosis/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "6cac610c-55c4-420d-afcb-87194e890006",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                             Description                  Fragment_Length\n",
      "   AP022614.1 AP022614.1 Mycobacterium parmense JCM 14742 DNA, nearly complete genome [199, 210, 306, 929, 1340, 1739]\n",
      "NZ_AP022614.1        NZ_AP022614.1 Mycobacterium parmense strain JCM 14742 chromosome [199, 210, 306, 929, 1340, 1739]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/parmense/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6c463f93-2dca-452b-a20c-861eece7aab5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                               Description                            Fragment_Length\n",
      "NZ_AP022573.1 NZ_AP022573.1 Mycobacterium saskatchewanense strain JCM 13016 chromosome, complete genome [123, 164, 225, 361, 396, 799, 1022, 1627]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/saskatchewanse/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = BsaWI.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = BsaWI.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "aa9faa93-183a-4eba-883b-fd30890a13aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                        Description             Fragment_Length\n",
      "NZ_AP022582.1 NZ_AP022582.1 Mycobacterium seoulense strain JCM 16018 chromosome, complete genome [164, 495, 798, 1420, 1859]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/seoulense/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = BsaWI.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = BsaWI.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "a435e3d4-ba9d-4ae0-aa9a-75f35d61852e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                Name                                                                                           Description                  Fragment_Length\n",
      "       NZ_AP018164.1                    NZ_AP018164.1 Mycobacterium shigaense strain JCM 32072 chromosome, complete genome [199, 209, 371, 929, 1343, 1650]\n",
      "       NZ_CP022927.1                          NZ_CP022927.1 Mycobacterium shigaense strain SCY chromosome, complete genome [199, 209, 371, 929, 1343, 1650]\n",
      "NZ_JAYDXR010000001.1 NZ_JAYDXR010000001.1 Mycobacterium shigaense strain 050038 contig00001, whole genome shotgun sequence [199, 209, 371, 929, 1343, 1650]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/Shiganse/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "2850a407-5a48-4b88-a909-6d3f99d9bfa5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                Name                                                                                                                                 Description              Fragment_Length\n",
      "   NZ_MVIK01000018.1 NZ_MVIK01000018.1 Mycobacterium shinjukuense strain CCUG 53584 Mycobacterium_shinjukuense_CCUG53584_contig18, whole genome shotgun sequence [110, 198, 1344, 1349, 1759]\n",
      "       NZ_AP022575.1                                                       NZ_AP022575.1 Mycobacterium shinjukuense strain JCM 14233 chromosome, complete genome [110, 198, 1344, 1349, 1759]\n",
      "NZ_JACKTO010000028.1          NZ_JACKTO010000028.1 Mycobacterium shinjukuense strain DSM 45663 MSHINJUKUENSEDSM45663_contig000028, whole genome shotgun sequence [110, 198, 1344, 1349, 1759]\n",
      "       NZ_OY970454.1                            NZ_OY970454.1 Mycobacterium shinjukuense strain DSM 45663 isolate human chromosome Mshi_45663, complete sequence [110, 198, 1344, 1350, 1759]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/shinjukuense/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "619dc990-ab2b-4d1a-a21e-4b5ecf313122",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                  Description        Fragment_Length\n",
      "   AP022572.1             AP022572.1 Mycobacterium shottsii JCM 12657 DNA, complete genome [452, 700, 1100, 2042]\n",
      "NZ_CP014860.2 NZ_CP014860.2 Mycobacterium shottsii strain M175 chromosome, complete genome [452, 700, 1100, 2042]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/shottsii/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "75096c1b-b4c0-4a49-8c68-3195fc26f3ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                      Description              Fragment_Length\n",
      "NZ_CP046600.1 NZ_CP046600.1 Mycobacterium spongiae strain FSD4b-SM chromosome, complete genome [211, 308, 1140, 1344, 1749]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/sponigae/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "5203f039-3656-45b8-9e95-86a1df0d5d03",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                          Description        Fragment_Length\n",
      "NZ_AP022587.1 NZ_AP022587.1 Mycobacterium stomatepiae strain JCM 17783 chromosome, complete genome [408, 929, 1340, 2034]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/stomatepiae/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "f7fa58af-bca7-433a-8de5-dbff4e093492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Name                                                                                           Description        Fragment_Length\n",
      "  NC_008611.1                                           NC_008611.1 Mycobacterium ulcerans Agy99, complete sequence [452, 892, 1348, 2046]\n",
      "NZ_AP017624.1 NZ_AP017624.1 Mycobacterium ulcerans subsp. shinshuense strain ATCC 33728 chromosome, complete genome [452, 892, 1348, 2042]\n",
      "NZ_CP083741.1                         NZ_CP083741.1 Mycobacterium ulcerans strain BS123 chromosome, complete genome [452, 892, 1348, 2042]\n",
      "NZ_CP085200.1                       NZ_CP085200.1 Mycobacterium ulcerans strain JKD8049 chromosome, complete genome [452, 892, 1348, 2043]\n",
      "NZ_CP092429.2                    NZ_CP092429.2 Mycobacterium ulcerans strain ATCC 19423 chromosome, complete genome [452, 892, 1348, 2043]\n",
      "NZ_LR135168.1                      NZ_LR135168.1 Mycobacterium ulcerans strain SGL03 isolate ITM032481 chromosome 1 [452, 892, 1348, 2046]\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Restriction import *\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# change path base to your device\n",
    "path = \"F:/dr ibrhim/ncbi/uclearns/ncbi_dataset/data\"\n",
    "\n",
    "P1 = \"TGCTTAACACATGCAAGTCGAACG\"\n",
    "P = Seq(\"GTTTCCCGCTTAGATGCTTTCAG\").reverse_complement()\n",
    "P2 = str(P)\n",
    "\n",
    "results = []   # store results for dataframe\n",
    "\n",
    "dir_list = os.listdir(path)\n",
    "\n",
    "for i in dir_list:\n",
    "    inpu = os.listdir(os.path.join(path, i))\n",
    "    for s in inpu:\n",
    "        if s.__contains__(i):\n",
    "            FASTAFILE = SeqIO.parse(os.path.join(path, i, s), 'fasta')\n",
    "            for fasta in FASTAFILE:\n",
    "                NAME, SEQUENCE, desc = str(fasta.name), str(fasta.seq), str(fasta.description)\n",
    "\n",
    "                # Forward strand\n",
    "                if P1 in SEQUENCE and P2 in SEQUENCE:\n",
    "                    subsequence = SEQUENCE[SEQUENCE.index(P1):SEQUENCE.index(P2) + len(P2)]\n",
    "                    try:\n",
    "                        sorttt = []\n",
    "                        g = Tsp45I.catalyze(Seq(subsequence))\n",
    "                        if len(g) < 2:\n",
    "                            continue\n",
    "                        for f in g:\n",
    "                            if len(f) not in sorttt and len(f) >= 50:\n",
    "                                sorttt.append(len(f))\n",
    "                        sorttt.sort()\n",
    "                        results.append({\n",
    "                                \"Name\": NAME,\n",
    "                                \"Description\": desc,\n",
    "                                \"Fragment_Length\": sorttt\n",
    "                            })\n",
    "                    except:\n",
    "                        pass\n",
    "\n",
    "                # Reverse complement strand\n",
    "                else:\n",
    "                    m = str(Seq(fasta.seq).complement())[::-1]\n",
    "                    if P1 in m and P2 in m:\n",
    "                        assemply_sequence = m[m.index(P1):m.index(P2) + len(P2)]\n",
    "                        try:\n",
    "                            sorttt = []\n",
    "                            g = Tsp45I.catalyze(Seq(assemply_sequence))\n",
    "                            if len(g) < 2:\n",
    "                                continue\n",
    "                            for f in g:\n",
    "                                if len(f) not in sorttt and len(f) >= 50:\n",
    "                                    sorttt.append(len(f))\n",
    "                            sorttt.sort()\n",
    "                            results.append({\n",
    "                                    \"Name\": NAME,\n",
    "                                    \"Description\": desc,\n",
    "                                    \"Fragment_Length\": sorttt\n",
    "                                })\n",
    "                        except:\n",
    "                            pass\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(results)\n",
    "\n",
    "# Print full table\n",
    "print(df.to_string(index=False))\n",
    "\n",
    "# Optional: save results\n",
    "# df.to_excel(\"restriction_results.xlsx\", index=False)\n",
    "# df.to_csv(\"restriction_results.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2aa59ed0-e873-4dfa-9a2b-222c861c8567",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
